File: src/Model/Offering.java
package Model;

public class Offering {
    private boolean isFull;
    private boolean isGroupOffering;
    private Type lessonType;

    public Offering(boolean isFull, boolean isGroupOffering, Type lessonType) {
        this.isFull = isFull;
        this.isGroupOffering = isGroupOffering;
        this.lessonType = lessonType;
    }

    public boolean isFull() {
        return isFull;
    }

    public boolean isGroupOffering() {
        return isGroupOffering;
    }

    public Type getLessonType() {
        return lessonType;
    }

    public void setFull(boolean full) {
        isFull = full;
    }

    public void setGroupOffering(boolean groupOffering) {
        isGroupOffering = groupOffering;
    }

    public void setLessonType(Type lessonType) {
        this.lessonType = lessonType;
    }

    public boolean equals(Offering aOffering) {
        return this.isFull == aOffering.isFull && this.isGroupOffering == aOffering.isGroupOffering &&
                this.lessonType == aOffering.lessonType;
    }
}

File: src/Model/Offering.py
from sqlalchemy import Column, Integer, String, Boolean, Enum
from sqlalchemy.orm import relationship
from Model.Type import Type
from database import Base

class Offering(Base):
    __tablename__ = 'offerings'

    id = Column(Integer, primary_key=True, autoincrement=True)
    is_full = Column(Boolean, nullable=False)  # Using Boolean for clarity
    lesson_type = Column(Enum(Type), nullable=False)
    is_group_offering = Column(Boolean, nullable=False)  # Added field for group offerings
    group_size = Column(Integer, nullable=False)  # Define group size

    # Relationship to Booking: An offering can have many bookings
    bookings = relationship("Booking", back_populates="offering")

    def __init__(self, is_full, lesson_type, is_group_offering, group_size):
        self.is_full = is_full
        self.lesson_type = lesson_type
        self.is_group_offering = is_group_offering
        self.group_size = group_size  # Use group_size in initialization

    def __repr__(self):
        return f"Offering(id={self.id}, is_full={self.is_full}, lesson_type={self.lesson_type}, is_group_offering={self.is_group_offering}, group_size={self.group_size})"
